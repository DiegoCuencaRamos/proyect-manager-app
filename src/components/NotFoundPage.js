import React, { Fragment } from 'react'
import { Link } from 'react-router-dom'
import Header from './Header'
import Title from './Title'

const NotFoundPage = () => (
    <Fragment>
        <Header />
        <Title 
            title='Page not found'
            description='The page you are looking for was not found'
        />
        <div className="container">
            <div id="goHomeContainer">
                <Link 
                    className="link"
                    to="/dashboard"
                >Go back to home page</Link>
            </div>
        </div>
    </Fragment>
)

export default NotFoundPage