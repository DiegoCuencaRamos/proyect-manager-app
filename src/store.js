import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store