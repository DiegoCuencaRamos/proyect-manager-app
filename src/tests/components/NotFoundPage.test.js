import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import store from '../../store'
import NotFoundPage from '../../components/NotFoundPage'

let wrapper

beforeEach(() => {
    wrapper = mount(
        <MemoryRouter initialEntries={[{ key: 'static' }]}>
            <Provider store={store}>
                <NotFoundPage />
            </Provider>
        </MemoryRouter>
    )
})

test('Should render NotFoundPage correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should navigate to dashboard page on "Go back to home page clicked"', () => {
    wrapper.find('#goHomeContainer Link').simulate('click', { button: 0 })
    expect(wrapper.find('Router').prop('history').location.pathname).toBe('/dashboard')
})