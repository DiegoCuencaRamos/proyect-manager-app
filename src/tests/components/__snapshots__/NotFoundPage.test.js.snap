// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NotFoundPage correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "static",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "static",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "static",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <NotFoundPage>
        <Header>
          <header
            className="header"
          >
            <div
              className="container"
            >
              <div
                className="header__content"
              >
                <Link
                  to="/dashboard"
                >
                  <LinkAnchor
                    href="/dashboard"
                    navigate={[Function]}
                  >
                    <a
                      href="/dashboard"
                      onClick={[Function]}
                    >
                      <span
                        className="header__title"
                      >
                        Proyect Manager
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
                <button
                  className="button--link"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
          </header>
        </Header>
        <Title
          description="The page you are looking for was not found"
          title="Page not found"
        >
          <div
            className="page-header"
          >
            <div
              className="container page-header__wrapper"
            >
              <div>
                <h1
                  className="page-header__title"
                >
                  Page not found
                </h1>
                <p
                  className="page-header__text"
                >
                  The page you are looking for was not found
                </p>
              </div>
            </div>
          </div>
        </Title>
        <div
          className="container"
        >
          <div
            id="goHomeContainer"
          >
            <Link
              className="link"
              to="/dashboard"
            >
              <LinkAnchor
                className="link"
                href="/dashboard"
                navigate={[Function]}
              >
                <a
                  className="link"
                  href="/dashboard"
                  onClick={[Function]}
                >
                  Go back to home page
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </NotFoundPage>
    </Provider>
  </Router>
</MemoryRouter>
`;
